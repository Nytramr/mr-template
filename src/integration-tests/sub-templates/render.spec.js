import { Template } from '../../index';

describe('Sub Templates', () => {
  describe('define and template actions', () => {
    test('simple define', () => {
      const rangeLoopTpl =
        '{{define "list"}}' + 'Print {{.}}' + '\n' + '{{end}}';

      const template = new Template('My template');
      expect(() => template.parse(rangeLoopTpl)).not.toThrow();

      let text = template.execute({});

      expect(text).toMatchSnapshot();
    });

    test('simple define and template', () => {
      const rangeLoopTpl =
        '{{define "hello"}}' +
        'Print: Hello world!' +
        '{{end}}' +
        '{{template "hello"}}';

      const template = new Template('My template');
      expect(() => template.parse(rangeLoopTpl)).not.toThrow();

      let text = template.execute({});

      expect(text).toMatchSnapshot();
    });

    test('template action with input data', () => {
      const rangeLoopTpl =
        '{{define "hello"}}' +
        'Print: Hello ' +
        '{{ .name }}' +
        '!' +
        '{{end}}' +
        '{{template "hello" .person }}';

      const template = new Template('My template');
      expect(() => template.parse(rangeLoopTpl)).not.toThrow();

      let text = template.execute({ person: { name: 'John' } });

      expect(text).toMatchSnapshot();
    });
  });

  describe('block action', () => {
    test('without input data', () => {
      const rangeLoopTpl =
        '{{block "hello"}}' + 'Print: Hello world!' + '{{end}}';

      const template = new Template('My template');
      expect(() => template.parse(rangeLoopTpl)).not.toThrow();

      let text = template.execute({});

      expect(text).toMatchSnapshot();
    });

    test('with input data', () => {
      const rangeLoopTpl =
        '{{block "hello" .person}}' +
        'Print: Hello ' +
        '{{ .name }}' +
        '!' +
        '{{end}}';

      const template = new Template('My template');
      expect(() => template.parse(rangeLoopTpl)).not.toThrow();

      let text = template.execute({ person: { name: 'Jenny' } });

      expect(text).toMatchSnapshot();
    });
  });
});
