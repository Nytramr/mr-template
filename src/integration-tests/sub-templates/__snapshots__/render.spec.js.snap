// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Templates block action with input data 1`] = `"Print: Hello Jenny!"`;

exports[`Sub Templates block action without input data 1`] = `"Print: Hello world!"`;

exports[`Sub Templates define and template actions simple define 1`] = `""`;

exports[`Sub Templates define and template actions simple define and template 1`] = `"Print: Hello world!"`;

exports[`Sub Templates define and template actions template action with input data 1`] = `"Print: Hello John!"`;
